sources/niUI/src/Canvas.cpp

```diff
@@ -597,6 +567,8 @@ struct sCanvasVGPathTesselatedRenderer : public cIUnknownImpl<iVGPathTesselatedR
         vertexColor.w *= apStyle->GetOpacity();
 
 #if 1
+        // we have the clear the ChannelTexture if there is any
+        mptrVGMaterial->SetChannelTexture(eMaterialChannel_Base, NULL);
```

Comment:
That seems dangerous, is there a place where the texture is set explicitly? A better way might be to create another material (for ex mptrVGMaterialGradient) and set the appropriate material based on the type of fill.